sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/MainView.groovy

```diff
@@ -0,0 +1,81 @@
+package life.qbic.portal.sampletracking.components
+
+
+import com.vaadin.ui.HorizontalLayout
+import com.vaadin.ui.Label
+import com.vaadin.ui.VerticalLayout
+import com.vaadin.ui.themes.ValoTheme
+import life.qbic.portal.sampletracking.components.projectoverview.ProjectOverviewView
+import life.qbic.portal.sampletracking.components.sampleoverview.SampleOverviewView
+import life.qbic.portal.sampletracking.components.toggle.ToggleButton
+
+/**
+ * <b><short description></b>
+ *
+ * <p><detailed description></p>
+ *
+ * @since <version tag>
+ */
+class MainView extends VerticalLayout {
+
+    private final ProjectOverviewView projectOverviewView
+    private final SampleOverviewView sampleOverviewView
+    private final HorizontalLayout hotbar = new HorizontalLayout()
+    protected final Label titleLabel = new Label("Sample Status Portlet")
+    private final ToggleButton projectSampleToggle
+
+    MainView(ProjectOverviewView projectOverviewView, SampleOverviewView sampleOverviewView) {
+        this.setMargin(false)
+
+        titleLabel.addStyleName(ValoTheme.LABEL_HUGE)
+
+        this.projectOverviewView = projectOverviewView
+        this.sampleOverviewView = sampleOverviewView
+        addHotbarItem(projectOverviewView.getHotbar())
+        addHotbarItem(sampleOverviewView.getHotbar())
```

Comment:
why extracting the hotbars here? I would expect each of the views (project or sample) to be "connected" to their hotbars

Reply:
They are extracted and injected into the overall hotbar here so that we don't have to have another element in the view and can preserve the existing look